internal/resources/policy/type/custom/input_schema.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package custom
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	policyrecipecustommodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/custom"
+	reciperesource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy/type/custom/recipe"
+)
+
+var (
+	inputSchema = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Input for the custom policy, having one of the valid recipes: tmc_block_nodeport_service, tmc_block_resources, tmc_block_rolebinding_subjects, tmc_external_ips, tmc_https_ingress or tmc_require_labels.",
+		Required:    true,
+		MaxItems:    1,
+		MinItems:    1,
+		ForceNew:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				reciperesource.TMCBlockNodeportServiceKey:     reciperesource.TMCBlockNodeportService,
+				reciperesource.TMCBlockResourcesKey:           reciperesource.TMCBlockResources,
+				reciperesource.TMCBlockRolebindingSubjectsKey: reciperesource.TMCBlockRolebindingSubjects,
+				reciperesource.TMCExternalIPSKey:              reciperesource.TMCExternalIps,
+				reciperesource.TMCHTTPSIngressKey:             reciperesource.TMCHTTPSIngress,
+				reciperesource.TMCRequireLabelsKey:            reciperesource.TMCRequireLabels,
+			},
+		},
+	}
+	recipesAllowed = [...]string{reciperesource.TMCBlockNodeportServiceKey, reciperesource.TMCBlockResourcesKey, reciperesource.TMCBlockRolebindingSubjectsKey, reciperesource.TMCExternalIPSKey, reciperesource.TMCHTTPSIngressKey, reciperesource.TMCRequireLabelsKey}
+)
+
+type (
+	recipe string
+	// InputRecipe is a struct for all types of custom policy inputs.
+	inputRecipe struct {
+		recipe                           recipe
+		inputTMCBlockNodeportService     *policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCCommonRecipe
+		inputTMCBlockResources           *policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCCommonRecipe
+		inputTMCBlockRolebindingSubjects *policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCBlockRoleBindingSubjects
+		inputTMCExternalIps              *policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCExternalIPS
+		inputTMCHTTPSIngress             *policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCCommonRecipe
+		inputTMCRequireLabels            *policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCRequireLabels
+	}
+)
+
+func constructInput(data []interface{}) (inputRecipeData *inputRecipe) {
```

Comment:
any reason why marshal and unmarshall cannot be used here ?

Reply:
Have not used that way anywhere else, and here we are following the style how security policies are implemented.